packages/cli/src/WorkflowExecuteAdditionalData.ts

```diff
@@ -516,7 +516,10 @@ function hookFunctionsSave(parentProcessMode?: string): IWorkflowExecuteHooks {
 					}
 
 					if (isManualMode && !saveManualExecutions && !fullRunData.waitTill) {
-						await Container.get(ExecutionRepository).softDelete(this.executionId);
+						await Container.get(ExecutionRepository).hardDelete({
+							workflowId: this.workflowData.id as string,
```

Comment:
From what I understand, In theory there can be no workflow without IDs anymore, so `IWorkflowBase['id']` should be `string` instead of `string | undefined` but that would be a larger change and I'd rather keep this PR focused.

Reply:
Yes this is some tech debt we'll have to address.